language/stdlib/modules/LibraConfig.move

```diff
@@ -100,38 +121,94 @@ module LibraConfig {
         LibraTimestamp::assert_genesis();
-        Roles::assert_libra_root(config_account);
+        Roles::assert_libra_root(lr_account);
         assert(
-            !exists<LibraConfig<Config>>(Signer::address_of(config_account)),
+            !exists<LibraConfig<Config>>(Signer::address_of(lr_account)),
             Errors::already_published(ELIBRA_CONFIG)
         );
-        move_to(config_account, LibraConfig { payload });
+        move_to(lr_account, LibraConfig { payload });
         ModifyConfigCapability<Config> {}
     }
+    spec fun publish_new_config_and_get_capability {
+        // TODO (dd): return value bug prevents this from working
+        // pragma opaque = true;
+        modifies global<LibraConfig<Config>>(CoreAddresses::LIBRA_ROOT_ADDRESS());
+        include LibraTimestamp::AbortsIfNotGenesis;
+        include Roles::AbortsIfNotLibraRoot{account: lr_account};
+        include AbortsIfPublished<Config>;
+        ensures global<LibraConfig<Config>>(CoreAddresses::LIBRA_ROOT_ADDRESS()) == LibraConfig { payload };
+        // TODO (dd) BUG: actual value has a dummy field but the generated spec does not.
+        // ensures result == ModifyConfigCapability<Config> {};
+    }
+    spec schema AbortsIfPublished<Config> {
+        aborts_if exists<LibraConfig<Config>>(CoreAddresses::LIBRA_ROOT_ADDRESS()) with Errors::ALREADY_PUBLISHED;
+    }
+
 
     // Publish a new config item. Only the config address can modify such config.
     // Does not trigger a reconfiguration. Will also publish the capability to
     // modify this config under the given account.
     public fun publish_new_config<Config: copyable>(
-        config_account: &signer,
+        lr_account: &signer,
         payload: Config
     ) {
-        let capability = publish_new_config_and_get_capability<Config>(config_account, payload);
+        let capability = publish_new_config_and_get_capability<Config>(lr_account, payload);
         assert(
-            !exists<ModifyConfigCapability<Config>>(Signer::address_of(config_account)),
+            !exists<ModifyConfigCapability<Config>>(Signer::address_of(lr_account)),
             Errors::already_published(EMODIFY_CAPABILITY)
         );
-        move_to(config_account, capability);
+        move_to(lr_account, capability);
+    }
+    spec fun publish_new_config {
+        pragma opaque = true;
+        // TODO (dd) BUG: the erroneous line caused a panic.
+        //   modifies global<Config>(CoreAddresses::LIBRA_ROOT_ADDRESS());
+        
+        // Everything has to be published at LIBRA_ROOT_ADDRESS because
+        // publish_new_config_and_get_capability aborts otherwise.
+        modifies global<LibraConfig<Config>>(CoreAddresses::LIBRA_ROOT_ADDRESS());
+        modifies global<ModifyConfigCapability<Config>>(CoreAddresses::LIBRA_ROOT_ADDRESS());
+        include PublishNewConfigAbortsIf<Config>;
+        // TODO (dd): the erroneous ensures below caused a panic.  Here is part of the stack trace:
```

Comment:
Interesting. That is worth an issue (every crash should). Feel free to keep it short and just point to here. You can also replace TODO(dd) with TODO(#<issue number>) if you open those issues before you merge. This may have the lowest boilerplate, as the actual description is easily discoverable by search.